### ISSUE TITLE
Terminal Size not detected on MacOS

### ISSUE THREAD

**Issue by AuroraYolo** [php-tui/php-tui#185]
There is no output on the page

**Comment by dantleech** [comment]
Need more information...

**Comment by AuroraYolo** [comment]
```php
<?php
declare(strict_types=1);

use PhpTui\Tui\DisplayBuilder;
use PhpTui\Tui\Extension\Core\Shape\MapResolution;
use PhpTui\Tui\Extension\Core\Shape\MapShape;
use PhpTui\Tui\Extension\Core\Widget\Block\Padding;
use PhpTui\Tui\Extension\Core\Widget\BlockWidget;
use PhpTui\Tui\Extension\Core\Widget\CanvasWidget;
use PhpTui\Tui\Extension\Core\Widget\GridWidget;
use PhpTui\Tui\Extension\Core\Widget\ParagraphWidget;
use PhpTui\Tui\Model\Direction;
use PhpTui\Tui\Model\Layout\Constraint;
use PhpTui\Tui\Model\Text\Text;
use PhpTui\Tui\Model\Text\Title;
use PhpTui\Tui\Model\Widget\Borders;
use PhpTui\Tui\Model\Widget\BorderType;

require 'vendor/autoload.php';

$display = DisplayBuilder::default()->build();
$display->clear();
$display->draw(
    GridWidget::default()
        ->direction(Direction::Horizontal)
        ->constraints(
            Constraint::percentage(50),
            Constraint::percentage(50)
        )
        ->widgets(
            BlockWidget::default()
                ->titles(Title::fromString('Left'))
                ->padding(Padding::all(2))
                ->borders(Borders::ALL)
                ->borderType(BorderType::Rounded)
                ->widget(
                    CanvasWidget::fromIntBounds(-180, 180, -90, 90)
                        ->draw(
                            MapShape::default()->resolution(MapResolution::High)
                        ),
                ),
            BlockWidget::default()
                ->titles(Title::fromString('Right'))
                ->padding(Padding::all(2))
                ->borders(Borders::ALL)
                ->borderType(BorderType::Rounded)
                ->widget(
                    ParagraphWidget::fromText(
                        Text::parse(<<<'EOT'
                            The <fg=green>world</> is the totality of <options=bold>entities</>,
                            the whole of reality, or everything that is.[1] The nature of the
                            world has been <fg=red>conceptualized</> differently in different fields. Some
                            conceptions see the world as unique while others talk of a
                            plurality of <bg=green>worlds</>.
                            EOT)
                    )
                ),
        )
);
```

```shell
 php tui.php 
PHP Warning:  Undefined array key 1 in /swow-admin/vendor/php-tui/php-tui/lib/term/src/InformationProvider/SizeFromSttyProvider.php on line 56

Warning: Undefined array key 1 in  /swow-admin/vendor/php-tui/php-tui/lib/term/src/InformationProvider/SizeFromSttyProvider.php on line 56
PHP Warning:  Undefined array key 2 in

```

# PHP Version
8.1

**Comment by dantleech** [comment]
which operating system do you have?

**Comment by dantleech** [comment]
but looks like         `if (false === preg_match('{rows ([0-9]+); columns ([0-9]+);}is', $out, $matches)) {` should be checking for `0` _or_ `false`

**Comment by dantleech** [comment]
https://github.com/php-tui/term/blob/4d9aad8095e5dfa10aa13dd0d159b36d4ca29046/src/InformationProvider/SizeFromSttyProvider.php#L52-L54

**Comment by AuroraYolo** [comment]
> which operating system do you have?
MacOs

**Comment by dantleech** [comment]
I've fixed the handling in term 0.3.1, you can do `composer update` - which should fix the _error_.

What happens when you run:

```
$ stty -a
```

?

it should return something like:

```
speed 38400 baud; rows 45; columns 143; line = 0;
intr = ^C; quit = ^\; erase = ^?; kill = ^U; eof = ^D; eol = <undef>; eol2 = <undef>; swtch = <undef>; start = ^Q; stop = ^S; susp = ^Z;
rprnt = ^R; werase = ^W; lnext = ^V; discard = ^O; min = 1; time = 0;
-parenb -parodd -cmspar cs8 -hupcl -cstopb cread -clocal -crtscts
-ignbrk -brkint -ignpar -parmrk -inpck -istrip -inlcr -igncr icrnl -ixon -ixoff -iuclc -ixany -imaxbel iutf8
opost -olcuc -ocrnl onlcr -onocr -onlret -ofill -ofdel nl0 cr0 tab0 bs0 vt0 ff0
isig icanon iexten echo echoe echok -echonl -noflsh -xcase -tostop -echoprt echoctl echoke -flusho -extproc
```

**Comment by AuroraYolo** [comment]
```shell
PHP Fatal error:  Uncaught RuntimeException: Could not determine terminal size! in /Users/heping/PhpWorkSpace/swow-cloud/swow-admin/vendor/php-tui/php-tui/src/Bridge/PhpTerm/PhpTermBackend.php:54
Stack trace:
#0 /Users/heping/PhpWorkSpace/swow-cloud/swow-admin/vendor/php-tui/php-tui/src/Model/Viewport/Fullscreen.php(20): PhpTui\Tui\Bridge\PhpTerm\PhpTermBackend->size()
#1 /Users/heping/PhpWorkSpace/swow-cloud/swow-admin/vendor/php-tui/php-tui/src/Model/Display/Display.php(39): PhpTui\Tui\Model\Viewport\Fullscreen->size(Object(PhpTui\Tui\Bridge\PhpTerm\PhpTermBackend))
#2 /Users/heping/PhpWorkSpace/swow-cloud/swow-admin/vendor/php-tui/php-tui/src/DisplayBuilder.php(134): PhpTui\Tui\Model\Display\Display::new(Object(PhpTui\Tui\Bridge\PhpTerm\PhpTermBackend), Object(PhpTui\Tui\Model\Viewport\Fullscreen), Object(PhpTui\Tui\Model\WidgetRenderer\AggregateWidgetRenderer))
#3 /Users/heping/PhpWorkSpace/swow-cloud/swow-admin/tui.php(17): PhpTui\Tui\DisplayBuilder->build()
#4 {main}


```

**Comment by dantleech** [comment]
yes I updaed my previous comment to indicate that you would get that exception can you run the command I suggested?

**Comment by AuroraYolo** [comment]
```shell
speed 9600 baud; 23 rows; 309 columns;
lflags: icanon isig iexten echo echoe -echok echoke -echonl echoctl
        -echoprt -altwerase -noflsh -tostop -flusho pendin -nokerninfo
        -extproc
iflags: -istrip icrnl -inlcr -igncr ixon -ixoff ixany imaxbel -iutf8
        -ignbrk brkint -inpck -ignpar -parmrk
oflags: opost onlcr -oxtabs -onocr -onlret
cflags: cread cs8 -parenb -parodd hupcl -clocal -cstopb -crtscts -dsrflow
        -dtrflow -mdmbuf
cchars: discard = ^O; dsusp = ^Y; eof = ^D; eol = <undef>;
        eol2 = <undef>; erase = ^?; intr = ^C; kill = ^U; lnext = ^V;
        min = 1; quit = ^\; reprint = ^R; start = ^Q; status = ^T;
        stop = ^S; susp = ^Z; time = 0; werase = ^W;
```

**Comment by AuroraYolo** [comment]
Please help me

**Comment by dantleech** [comment]
should be fixed with latest tag (0.3.2) can you confirm?

**Comment by AuroraYolo** [comment]
ok!
